examples/lock-app/esp32/main/LEDWidget.cpp

```diff
@@ -61,9 +61,9 @@ void LEDWidget::Animate()
 {
     if (mBlinkOnTimeMS != 0 && mBlinkOffTimeMS != 0)
     {
-        int64_t nowUS            = ::chip::System::Clock::GetMonotonicMicroseconds();
-        int64_t stateDurUS       = ((mState) ? mBlinkOnTimeMS : mBlinkOffTimeMS) * 1000LL;
-        int64_t nextChangeTimeUS = mLastChangeTimeUS + stateDurUS;
+        uint64_t nowUS            = chip::System::SystemClock().GetMonotonicMicroseconds();
+        uint64_t stateDurUS       = ((mState) ? mBlinkOnTimeMS : mBlinkOffTimeMS) * 1000LL;
+        uint64_t nextChangeTimeUS = mLastChangeTimeUS + stateDurUS;
 
         if (nowUS > nextChangeTimeUS)
```

Comment:
This method of "scheduling" is not wraparound-safe. This is "OK" since we have 64 bit counters, but if this appears on any 32 bit monotonic clock in ms, it is unsafe after ~49 days.

Reply:
Indeed; I didn't really change any existing consumers beyond `Clock::` ↦ `SystemClock().`. We do have an `IsEarlier()` (which assumes an interval under half the type range) but can't catch people using plain integer operations without a type-safe tick, which is tempting, but would be a separate issue. (Now that I look at `IsEarlier()`, it does need a little immediate work, namely a static_assert that the type is unsigned for the ‘half range’ arithmetic .)

Reply:
This is LEDWidget example code. I believe fixing is outside of the scope of this PR. Can fix it separately if we feel it is important  enough to fix.